frame/support/src/storage/hashed.rs

```diff
@@ -72,6 +80,8 @@ where
 	HashFn: Fn(&[u8]) -> R,
 	R: AsRef<[u8]>,
 {
+	sp_tracing::event!(target: "frame_storage",
+		sp_tracing::Level::INFO, fun="put", key=?key, value=?value.encode());
```

Comment:
this (and the other changes here) are what we were actually aiming for: have wasm tell us the raw key and values to trace state changes. \o/

Reply:
All these stuff here is already traced on the host. I don't understand why we need to trace this again :shrug:

Reply:
There's two main differences: this doesn't trace the same way as the host-function does – with opening a span and recording how long it takes  – but merely fires an event that a specific key is being written to, providing – that's the second difference – the _unhashed_ key to it. To the host function there is just a key being written to, it doesn't know the underlying information that lead to that key being picked as they are runtime/FRAME specific. That information is provided through this `event!`

Reply:
And what happns when you use `unhashed` directly?

Reply:
This PR doesn't add these changes yet, but you are right, we probably should add traces to `map`, `double_map` and alike to offer similar support of knowing the yet-to-be-hashed keys. Not sure if we want add that into this PR, too, or in a follow-up (arguably, the changes in this file could also be out of scope from the name of the PR) – maybe there is a way to do that in more generic way. If you prefer I can remove these changes from this PR and we move that entire concept there.

Reply:
Yeah, probably better to add these changes later in another pr to make them better reviewable.